I added support to run cantact-fw on STM32F072B-DISCO and build files for MDK-ARM/Keil

trollcop opened the pull request:
Since STM32F072B-DISCO is kinda ubiquitous and can be had for $12 I've added support to run cantact-fw on this board.
An external canbus transceiver breakout can be connected to VDD/PB8/PB9/GND pins on the -DISCO easily. No soldering or modifications needed on the board.

During making the port, I made LED pins configurable, fixed some compile/style warnings and properly fixed running off internal oscillator (072B-DISCO doesn't have a crystal mounted).

I don't have a linklayer board to test with but the changes are pretty simple and should not affect operation on original board.

Oh, and I added build files for MDK-ARM v4, shouldn't hurt anyone I suppose.

trollcop committed:
Remove unnecessary < 0 comparison for uint32_t

trollcop committed:
Add support for running on STM32F072B-DISCO
Made LED pins/GPIO configurable instead of hardcoded.
Corrected HSI48 init code (did not work out of the box on -disco, didn't bother investigating why, just replaced with working code)

trollcop committed:
Add DISCO docs

jxltom commented:
1. It might be better if makefile is also updated based on stm32f072
2. It is also better to update startup_*.s file
3. It will be nice to support two led with different port such as GPIOB and GPIOC

trollcop committed:
Updated Makefile to build for cantact/disco boards (build with make BULID_HW=disco, default cantact, if no arguments specified)
Added (well, actually copied from CMSIS template folder) startup/linker files for stmf072 for gcc build
Made LED GPIO ports selectable for each LED (still need to edit LED_GPIO_CLK by the user but at that point they should know what they're doing)

trollcop commented:
Allright, I added a new commit taking care of those requests (I don't use gcc/Makefile but I tested both builds and they work).
I'm not sure why its not showing up in the list here but the last commit is here: https://github.com/trollcop/cantact-fw/commit/abca3389b29b7642b808d6601ad96e289097af96

can be invoked as just `make` for cantact default, or `make BUILD_HW=disco` for the -discovery build.

jxltom commented:
This is nice.

Just one more thing, maybe it is better to define seperate ```LED_RED_GPIO_CLK``` and ```LED_GREEN_GPIO_CLK``` to support different ports?

trollcop commented:
> This is nice.
> 
> Just one more thing, maybe it is better to define seperate `LED_RED_GPIO_CLK` and `LED_GREEN_GPIO_CLK` to support different ports?

I've left a comment in the PR regarding that - there's no simple way other than duplicating the define, and at the point where a user need to change this they probably know what they're doing. If you have a suggestion of how to implement it better than something like 

```
#define LED_GREEN_GPIO_CLK    __GPIOB_CLK_ENABLE
#define LED_RED_GPIO_CLK    __GPIOB_CLK_ENABLE
```

I'm all ears, or else it can just be made that way.

trollcop committed:
Added LED GPIO CLK to configuration as well.
